idp/idputil/idputil.go

```diff
@@ -254,3 +269,308 @@ func CookiePathRelativeToLocation(cookiePath, location string, skipLocation bool
 	}
 	return u.Path + cookiePath
 }
+
+// JsonResponse mashals the provided to JSON and writes it
+// to the provided response writer.
+func JsonResponse(w http.ResponseWriter, data interface{}, c int) {
+	dj, err := json.Marshal(data)
+	if err != nil {
+		http.Error(w, "failed to create a json response", http.StatusInternalServerError)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(c)
+	fmt.Fprintf(w, "%s", dj)
+}
+
+// WebAuthnUserFromIdentity converts the provided store Identity to
```

Comment:
I think I would be inclined to put the webauthn stuff into it's own file, or possibly even package.

Reply:
done